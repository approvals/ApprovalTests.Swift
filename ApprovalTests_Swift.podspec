Pod::Spec.new do |s|
  s.name         = 'ApprovalTests_Swift'
  s.version      = '2.0.0'
  s.summary      = 'ApprovalTests for Swift, a powerful alternative to assertions'

  s.description = <<-DESC
                    Approval Tests are an alternative to assertions. Youâ€™ll find them useful for
                    testing objects with complex values (such as long strings), lots of properties,
                    or collections of objects.

                    ApprovalTests.Swift is compatible with the XCTest testing framework.
                  DESC

  s.homepage = 'https://github.com/approvals/ApprovalTests.Swift'
  s.license  = { :type => 'Apache License, Version 2.0', :file => 'LICENSE' }
  s.author   = { 'Jon Reid' => 'jon@qualitycoding.org' }
  s.social_media_url = 'https://twitter.com/qcoding'

  s.swift_version = '5.0'
  s.ios.deployment_target = '12.0'
  s.osx.deployment_target = '10.15'

  s.source = { :git => 'https://github.com/approvals/ApprovalTests.Swift.git', :tag => 'v2.0.0' }
  s.framework = 'XCTest'
  s.source_files      = "ApprovalTests.Swift/iOSApprovalsWatcher.py", "ApprovalTests.Swift/**/*.swift" 
  s.ios.exclude_files = "ApprovalTests.Swift/Reporters/macOS/*.swift"
  s.osx.exclude_files = "ApprovalTests.Swift/Reporters/iOS/*.swift"

end
