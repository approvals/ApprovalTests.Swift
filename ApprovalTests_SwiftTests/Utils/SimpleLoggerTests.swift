#if os(iOS)
  import ApprovalTests_iOS
#else
  import ApprovalTests_Swift
#endif
import XCTest

final class SimpleLoggerTests: XCTestCase {
    func test() throws {
        let output = SimpleLogger.logToString()
        do {
            let m = SimpleLogger.useMarkers()
            SimpleLogger.printLineNumber()
            do {
                let m2 = SimpleLogger.useMarkers()
                let name = "llewellyn"
                SimpleLogger.variable("name", name)
                for _ in 0 ..< 142 {
                    SimpleLogger.hourGlass()
                }
            }
        }
        try Approvals.verify(output)
    }

    func test_timestamps() throws {
        let output = SimpleLogger.logToString()
        SimpleLogger.timestamp = true
        var dates: [Date] = [
            Date(timeIntervalSince1970: 0),
            Date(timeIntervalSince1970: 0.5),
        ] 
        SimpleLogger.timer = {
            let first = dates.first
            dates = Array(dates.dropFirst())
            return first!
        }
        SimpleLogger.event("1")
        SimpleLogger.event("2")
        try Approvals.verify(output)
    }

    func test_markers() throws {
        let output = SimpleLogger.logToString()
        add(1, 2)
        try Approvals.verify(output)
    }

    func test_variable() throws {
        let output = SimpleLogger.logToString()
        SimpleLogger.variable("name", "Jon")
        SimpleLogger.variable("name", "Jon", showTypes: true)
        let subclasses = [ExtendA(), ExtendB()]
        SimpleLogger.variable("subclasses", subclasses)
        SimpleLogger.variable("subclasses", subclasses, showTypes: true)
        let mix: [Any] = [1, false, "fred"]
        SimpleLogger.variable("mix", mix, showTypes: true)
        try Approvals.verify(output)
    }

    private func add(_ a: Int, _ b: Int) {
        do {
            let m = SimpleLogger.useMarkers("\(a), \(b)")
        }
    }
}

class Base {}
class ExtendA: Base, CustomStringConvertible {
    private(set) var description: String = "ExtendA"
}
class ExtendB: Base, CustomStringConvertible {
    private(set) var description: String = "ExtendB"
}

/*
 public void test()
 {
     StringBuffer output = SimpleLogger.logToString();
     try (Markers m = SimpleLogger.useMarkers();)
     {
         try (Markers m2 = SimpleLogger.useMarkers();)
         {
             SimpleLogger.event("Starting Logging");
             String name = "llewellyn";
             SimpleLogger.variable("name", name);
             SimpleLogger.query("Select * from people");
             for (int i = 0; i < 42; i++)
             {
                 SimpleLogger.hourGlass();
             }
             SimpleLogger.variable("Numbers", new Integer[]{1, 2, 3, 4, 5});
             SimpleLogger.warning(new Error());
         }
     }
     Approvals.verify(output);
 }
 */
