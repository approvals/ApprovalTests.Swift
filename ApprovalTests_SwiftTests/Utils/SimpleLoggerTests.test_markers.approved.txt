
-> in: add(1, 2) in SimpleLoggerTests.swift
<- out: add(_:_:)
